bin/check_replication.rb

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/ruby -w
+
+require 'rest-client'
+require 'json'
+
+def replication_status(u,p,src,dst)
+	src = "http://#{src}"
+	dst = "http://#{dst}"
+	check_replication_status = RestClient.get("#{u}:#{p}@localhost:5984/_active_tasks")
+  status = JSON.parse(check_replication_status)
+  if status.size == 0 then
+  	begin
+		 	rep_id = RestClient.post("http://#{u}:#{p}@localhost:5984/_replicate","{\"source\":\"#{src}\",\"target\":\"#{dst}\", \"continuous\":true}",content_type: :json)
+		 	$rep_id = JSON.parse(rep_id)
+		 	$rep_id = $rep_id['_local_id']
+		rescue => e
+			puts e.response
+		end	
+		 	$num += 1
+  else
+  	repli_id = status.map{|v|v['replication_id']}
+	  if !repli_id.include?("#{$rep_id}") then
+	  	 #start replication
+		  	begin
+		    	rep_id = RestClient.post("http://#{u}:#{p}@localhost:5984/_replicate","{\"source\":\"#{src}\",\"target\":\"#{dst}\", \"continuous\":true}",content_type: :json)
+		    	$rep_id = JSON.parse(rep_id)
+		    	$rep_id = $rep_id['_local_id']
+				rescue => e
+					puts e.response
+				end	
+		   	$num += 1
+		end
+	end
+end
+
+u = ARGV[0]
+p = ARGV[1]
+src = ARGV[2]
+dst = ARGV[3]
+
+#Initialize variables
+i = 0
+a = ['|','/','-','\\']
+n = 0
+$num = 0
+$rep_id = ""
+
+while i != -1
+	sleep (1)
+	replication_status(u,p,src,dst)
```

Comment:
Isn't the replication bi-directional? Shouldn't we have a second call to replication_status with src and dst flipped for the reverse replication?

```ruby
replication_status(u, p, src, dst)
replication_status(u, p, dst, src)
```

Reply:
Yes we are supposed to have as call which inverts the source and destination will update accordingly but with the suggestion of having it run and the central we will need to put it in nlims at the central location.